### PR TITLE
[ci/release] remove legacy release tests

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This PR removes unused legacy release tests and release test files. They've been usually migrated to one of the nightly/daily test suites.

The one test suite still left here is the `kubernetes_tests` directory. cc @DmitriGekhtman can this be removed? Is this tested elsewhere? Or should this somehow be incorporated into the release process?

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by DmitriGekhtman** [comment]
You can remove the Kubernetes directory.

**Review by rkooo567 — APPROVED** [review]

File: release/data_processing_tests/dask-on-ray-test.sh [hunk 1]
```diff
@@ -1,43 +0,0 @@
...
-pip uninstall -y ray
-pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/releases/1.5.0/ddad6a2458bded21035797ec84ba67b8a8c15ca5/ray-1.5.0-cp37-cp37m-macosx_10_13_intel.whl
-
-for dask_version in "${DASK_VERSION[@]}"
-do   # The quotes are necessary here
-    echo "=================================================="
-    echo "Downloading Dask of version '${dask_version}'"
-    pip uninstall -y dask
-    pip install -U "dask[complete]==${dask_version}"
-    printf "==================================================\n\n\n"
```
Comment (rkooo567) [inline release/data_processing_tests/dask-on-ray-test.sh:36]: Don't we still need it? It is not automated now afaik?

Comment (krfricke) [reply release/data_processing_tests/dask-on-ray-test.sh:36]: It was not used in the release process. I'll open an issue to track this

Comment (krfricke) [reply release/data_processing_tests/dask-on-ray-test.sh:36]: Tracked here https://github.com/ray-project/ray/issues/18628

**Pull request merged**